src/modal.component.ts

```diff
@@ -21,20 +34,25 @@ export class ModalComponent implements OnDestroy {
   }
 
   open(): void {
-    document.body.style.overflow = 'hidden';
+    this.modalService.setModal(this);
+    document.body.classList.add('modal-open');
 
     this.visible = true;
+    setTimeout(() => {
```

Comment:
Animation looks like a good feature to me. I will recheck the implementation later, but I cannot see any major issues here right now. :+1:

Reply:
I'm still learning of angular way to do animation. I think it will be better to do with angular approach, but for now, this looks good. I've taken this from the example on stackoverflow. also, the template has it, but it was missing the script code.